Add basic components

matheus-fls opened the pull request:
Add the basic React components (App, Display, Button and ButtonPanel). No styling yet.

matheus-fls committed:
Add basic components

matheus-fls committed:
Add result prop to display

matheus-fls committed:
Render ugly buttons

matheus-fls committed:
Remove comment

matheus-fls committed:
Refactor components and add state

matheus-fls committed:
Fix linter errors

matheus-fls committed:
Refactor display and fix errors

matheus-fls committed:
Fix linter errors

matheus-fls committed:
Add new key to button panel div

matheus-fls committed:
Fix linter errors

Oghenebrume50 commented on src/components/ButtonPanel.js:
```diff
@@ -0,0 +1,21 @@
+  return (
+    <div>
+      {buttons.map(e => (
+        <div key={(Math.random() * 100).toString()}>
```
using a random value a the key for a component is counter productive to the idea of keys, please see this [link](https://stackoverflow.com/questions/29808636/when-giving-unique-keys-to-components-is-it-okay-to-use-math-random-for-gener)

Oghenebrume50 commented on src/components/Display.js:
```diff
@@ -0,0 +1,9 @@
+      <h1>{state.result}</h1>
+    </div>
+  );
+}
```
please validate that result is a string ![screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/validateAsString.png)

Also use the defaultprop value in react to assign a defualt value to the result prop 
![screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/DefaultProp.png)

Oghenebrume50 reviewed (changes requested):
### STATUS: CHANGES REQUIRED

Hi @matheus-fls good start to this project, everything is coming along well, :+1:  please make changes as outline below and result for a re-review 

**RAPHAEL NORIODE** :sunglasses:

matheus-fls committed:
Change ButtonPanel keys, add default prop value to display

matheus-fls committed:
Add default props to display

matheus-fls committed:
Fix linter errors

matheus-fls commented:
Thank you very much for your excellent review, Raphael!

I believe all the required changes were made.

Submitting for another review.

bolah2009 commented on .eslintrc.json:
```diff
@@ -16,6 +16,7 @@
     "import/no-unresolved": "off",
-    "no-shadow": "off"
+    "no-shadow": "off",
+    "react/no-array-index-key": 0
```
Please let's use the **correct config files**.

![image](https://user-images.githubusercontent.com/36057474/71924065-8b46a000-318e-11ea-8f37-02c188bde393.png)

See https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/

If there's a reason why the config file needs to be changed, let's discuss it on this PR.

bolah2009 commented on src/components/Button.js:
```diff
@@ -0,0 +1,14 @@
+      <button type="button">{ name }</button>
+    </>
+  );
+}
```
[Optional]

Consider using ES6 arrow function instead. With arrow function, you don't need an implicit `return` keyword for this component.


```jsx
export default Button = ({ name }) => (
      <button type="button">{ name }</button>
 );
```

Also, is there a reason why you are using an empty tag in this component? :thinking:

bolah2009 commented on src/components/ButtonPanel.js:
```diff
@@ -0,0 +1,21 @@
+      ))}
+    </div>
+  );
+}
```
[Optional]

I think using a random number may be better than using the index of the array since we have less than 10 lists hence the linter warnings. 

You can disable the linter warning using inline comments instead of changing the config file, see https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments

Alternatively, you could also use a module that generates a random string, see https://www.npmjs.com/package/uuid

bolah2009 reviewed (changes requested):
#### Hi @matheus-fls 

Good job with the suggested changes. 

It seems you changed the config while trying to make one of the optional changes, you can revert to the way the code was or see other suggestions in the comments below.

Please, feel free to ignore the comments tagged as `Optional`.

#### :recycle:  Make these changes and submit another code review request. 👍🏾

#### Feel free to reach me on slack if you having issues implementing these features. :relaxed: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

matheus-fls committed:
Remove empty tag from Button, reverse eslint config to original, disable rule inline on ButtonPanel

matheus-fls committed:
Fix linter errors

matheus-fls commented:
Hello Bola! Thank you very much for your great review!

I just wanted to point out, regarding using the index to name the keys: I understand that the rule makes a lot of sense for dynamic arrays, but in this case, the buttons array will not change, so I'm naming the button rows using the array index + 1 because I think it looks good and organized. And like Raphael pointed out in his review, using random names might be counterproductive. 

I reverted the eslint file back to the original and disabled the alert using an inline comment. I also removed the empty tag from the Button file because it was really pointless to use it there. 

Regarding your suggestion to use arrow functions, I tried to use it but I had some issues, so I'll just stick with the regular function for now.

Thank you very much for your suggestions, I did my best to do the changes as you required.

I will now submit the code for another review.

bolah2009 reviewed (approved):
It's a 👍🏾 from me!

### :heavy_check_mark: Approved!

Nice job implementing the suggested changes.

Good luck with the next milestones. :star2: 

Cheers
[Bola](https://bolabuari.com/)
